### DISCUSSION TITLE
Loadbalancer (F5) Info

### DISCUSSION THREAD

**Discussion by mhabiballa** [netbox-community/netbox#11134] [Q&A]
Dears @ Netbox
Thank you very much for this amazing tool. Been using it for 3 months now in a "big" DC/DR env. Initially, deployed it in my own PC as ab IPAM just to grasp and get full view of this DC/DR sites, with far target to help me really doc and understand how things are moving there.
3 months later, it became a server with different levels access starting from NOC teams up to Infra Mgrs. :)

Now, I've been asked by F5 team, if they could upload their loadbalancers infos to it to be linked to the F5Devic/IP using IPAM module. They proposed the following table to be accessed from F5 device in netbox :
**Seq (no.) | URL | vIP | Port | Pool_Name | Pool_Members(IPs) | F5 Device info (DNS-Hostname) | Location | Ticket no. | Owner | Contact info |** Comments
Is this possible to be done in netbox IPAM? Or to be linked to IPAM-IP addresses somehow? Is it even possible to do?

Short story: Is it possible to have a customized table fro F5 devices that contains different kind of loadbalancing infos as stated above?
Thank you,

**Comment by mtinberg** [comment]
I'm not sure an export template can get you all the data you want because I don't think they can look up other (related) data models in the Django ORM than the one you are exporting, but maybe a Script (or Report) could return data in a format your F5 can use (modeling what you want in custom fields on /ipam/prefix/, /ipam/ip-address/ and /dcim/device/) or an external script that uses the API to get the data in a format you can use.  The benefit of using a Script in Netbox is managing the code in one place and using the Django ORM directly to craft the query you want, the benefit of using the REST API is that it's simple and well documented on how to integrate with other tools/languages and the REST API handlers pre-query related fields and build a useful view of data without you needing to re-create that yourself.

In either case, if you are storing more complex relationships (pool(text), poll_membership(multi-object ip-address reference), ticket(text or integer), etc.) you probably want a Script to provide a form with all the options/fields that are mandatory to fill out and a Report to validate that existing records are consistent with your business rules, so that the automation isn't working on bad/incomplete data and potentially making half-baked changes.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: mhabiballa ***@***.***>
Sent: Thursday, December 8, 2022 3:07 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Loadbalancer (F5) Info (Discussion #11134)


Dears @ Netbox
Thank you very much for this amazing tool. Been using it for 3 months now in a "big" DC/DR env. Initially, deployed it in my own PC as ab IPAM just to grasp and get full view of this DC/DR sites, with far target to help me really doc and understand how things are moving there.
3 months later, it became a server with different levels access starting from NOC teams up to Infra Mgrs. :)

Now, I've been asked by F5 team, if they could upload their loadbalancers infos to it to be linked to the F5Devic/IP using IPAM module. They proposed the following table to be accessed from F5 device in netbox :
Seq (no.) | URL | vIP | Port | Pool_Name | Pool_Members(IPs) | F5 Device info (DNS-Hostname) | Location | Ticket no. | Owner | Contact info | Comments

Is this possible to be done in netbox IPAM? Or to be linked to IPAM-IP addresses somehow? Is it even possible to do?

Thank you,
